comp.go

```diff
@@ -74,15 +79,12 @@ func equals(a, b any) bool {
 		return a == b
 	}
 
-	if isNumber(a) {
-		return isPrimitive(b) && toNumber(a) == toNumber(b)
+	if isPrimitive(a) && isPrimitive(b) && (isNumber(a) || isNumber(b)) {
+		return toNumberForLess(a) == toNumberForLess(b)
```

Comment:
why did you use `toNumberForLess` here instead of `toNumber`?

Reply:
b/c when a or b is a number but the other isn't the other gets coerced into a number according to the same rules you already had in toNumberForLess

Reply:
I'll think about a new name for `toNumberForLess` because it's causing confusion on my mind. But I got the point.

Reply:
maybe `toNumberForComparison`?